index.d.ts

```diff
@@ -45,6 +46,10 @@ export namespace Sub {
   function none(): Sub;
 }
 
+export function withSubscriptions<Props>(
+  UserComponent: React.ComponentType<Props>,
+  subscriptions: (model: any) => Sub): React.ComponentType<Props>;
```

Comment:
Could we have piped through a `T` here?

Reply:
I think so, along with enforcing that the type of the model matches the type of the model on the props. Let me try that